hw1/ex2/ex2.py

```diff
@@ -0,0 +1,176 @@
+import psutil
+import time
+import argparse
+import uuid
+import redis
+
+
+class RedisDB:
+    def __init__(self, host: str,
+            port: int,
+            user: str,
+            password: str):
+        self.host = host
+        self.port = port
+        self.user = user
+        self.password = password
+        self.redis_client = None
+        self.connected = False
+
+    def connect(self) -> bool:
+        if self.redis_client is None:
+            self.redis_client = redis.Redis(
+                host=self.host,
+                username=self.user,
+                port=self.port,
+                password=self.password
+                )
+            self.connected = self.redis_client.ping()
+            return self.connected
+        return False
```

Comment:
Why return false if there is a connected client?

Reply:
Because a new connection is not opened. To check if the connection is present use is_connected method instead.